sql-plugin/src/main/java/com/nvidia/spark/rapids/GpuColumnVector.java

```diff
@@ -303,12 +295,40 @@ public HostColumnVector[] buildHostColumns() {
       }
     }
 
+    /**
+     * Build a columnar batch without releasing the holding data on host.
+     * It is safe to call this multiple times, and data will be released
+     * after a call to `close`.
+     */
+    public ColumnarBatch tryBuild(int rows) {
+      if (hostColumns == null) {
+        hostColumns = buildHostColumns();
+      }
+      return build(rows, i -> hostColumns[i].copyToDevice());
+    }
+
     @Override
     public void close() {
       for (ai.rapids.cudf.HostColumnVector.ColumnBuilder b: builders) {
         if (b != null) {
-          b.close();
+          try {
+            b.close();
+          } catch (Throwable e) {
+            /* ignore the exception */
```

Comment:
this, and the next catch seem a bit odd. What exception were you seeing? 

Should this instead use something like `safeClose()` and produce a single exception after trying to close all, and returning a single exception object that has the individual exceptions supressed?

Reply:
I just want to make sure it will always try to close the HostColumns no matter any exception happens when closing the builders. Something likes
```
      try {
        for (ai.rapids.cudf.HostColumnVector.ColumnBuilder b: builders) {
          if (b != null) {
            b.close();
          }
        }
      } finally {
        if (hostColumns != null) {
          for (ai.rapids.cudf.HostColumnVector hcv: hostColumns) {
            if (hcv != null) {
              hcv.close();
            }
          }
          hostColumns = null;
        }
      }
```

Is this OK ? @abellina @revans2, or do you want me to do as the following. Java requires explicit declaration for non runtime exceptions, we only need to catch the runtime exceptions here.
```
      RuntimeException toThrow = null;
      for (ai.rapids.cudf.HostColumnVector.ColumnBuilder b: builders) {
        if (b != null) {
          try {
            b.close();
          } catch (RuntimeException e) {
            if (toThrow == null) {
              toThrow = e;
            } else {
              toThrow.addSuppressed(e);
            }
          }
        }
      }
      if (hostColumns != null) {
        for (ai.rapids.cudf.HostColumnVector hcv: hostColumns) {
          if (hcv != null) {
            try {
              hcv.close();
            } catch (RuntimeException e) {
              if (toThrow == null) {
                toThrow = e;
              } else {
                toThrow.addSuppressed(e);
              }
            }
          }
        }
        hostColumns = null;
      }
      if (toThrow != null) {
        throw toThrow;
      }
    }
```